src/tests/e2etest/HostNameTests.cs

```diff
@@ -0,0 +1,115 @@
+using System.Text.Json;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+namespace e2etesting
+{
+    public class HostNameTests : E2eTest
+    {   
+        string ComponentName = "HostName";
+        public partial class HostName
+        {
+            public string Name { get; set; }
+            public string Hosts { get; set; }
+        }
+        public partial class DesiredHostName
+        {
+            public string DesiredName { get; set; }
+            public string DesiredHosts { get; set; }
+        }
+
+        public partial class Connection
+        {
+            public string ConnectionState { get; set; }
+        }
+        public partial class ExpectedHostNamePattern
+        {
+            public Regex NamePattern { get; set; }
+            public Regex HostsPattern { get; set; }
+        }
+        HostName deserializedReportedObject;
+
+        [Test]      
+        public void HostNameTest_Get()
+        {
+            var expectedHostNamePattern = new ExpectedHostNamePattern
+            {
+                NamePattern = new Regex(@"(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])"),
+                HostsPattern = new Regex(@"(((([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9][0-9]|2[0-4][0-9]|25[0-5]))|((([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|[fF][eE]80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::([fF][eE]{4}(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))))( +((([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\\-]*[a-zA-Z0-9])\\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\\-]*[A-Za-z0-9])))+")
+            };
+
+            if ((GetTwin().ConnectionState == DeviceConnectionState.Disconnected) || (GetTwin().Status == DeviceStatus.Disabled))
+            {
+                Assert.Fail("Module is disconnected - HostNameTest_Get");
+            }
+
+            deserializedReportedObject = JsonSerializer.Deserialize<HostName>(GetTwin().Properties.Reported[ComponentName].ToString());
+
+            Assert.True(IsRegexMatch(expectedHostNamePattern.NamePattern, deserializedReportedObject.Name));
+            Assert.True(IsRegexMatch(expectedHostNamePattern.HostsPattern, deserializedReportedObject.Hosts));
+        }
+
+        [Test]
+        public void HostNameTest_Set_Get()
+        {
+            int extendedTimeoutSeconds = 70;
+            SetTwinTimeoutSeconds(extendedTimeoutSeconds);
```

Comment:
This will affect all tests and will override the default value that is specified as a parameter in the pipeline. The timeout should only be changed for this test.
I would recommend looking at [SetUp](https://docs.nunit.org/articles/nunit/writing-tests/attributes/setup.html)/TearDown so we can set/reset the value back to the original default for the rest of the tests. You will have to create your own _TestFixture_ to do this.

Reply:
You might want to consider not hardcoding the timeout but maybe using a multiple of the default value that was originally specified if the intent is to have the test timeout longer. But why exactly are we bumping the timeout for this test? Does setting the hostname a long-running operation?